### DISCUSSION TITLE
Stop Rebasing open renovate MR when MR is in Merge Train

### DISCUSSION THREAD

**Discussion by Preisschild** [renovatebot/renovate#29842] [Suggest an Idea]
### Tell us more.

If you add a renovate MR to a Gitlab Merge Train and another MR is merged (for example, if the MR train position is before the other renovate MR) renovate will rebase the MR that is already in the merge train.

This causes Gitlab to remove the MR from the Merge Train because new commits were added, which means that the MR needs to be added to the Merge Train by a reviewer again.

I think adding a feature to not rebase MRs when they are in a merge train would make using renovate with Gitlab Merge Trains better.

**Comment by rarkins** [comment]
Have you tried rebaseWhen=conflicted ? If so please attach the debug logs from when that is enabled but Renovate rebases against your wishes

**Reply by Preisschild** [reply]
I guess `rebaseWhen=conflicted` works too :)

I forgot that you dont really need to have your MRs rebased when using Merge Trains

Thank you :)
